### PR TITLE
Configured Sync to allow multiple targets

### LINKED ISSUE #3261: Allow sync to multiple targets

### PR DESCRIPTION
**Pull request by RafaPolit**
fixes #3261

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

File: app/api/sync/syncWorker.js [hunk 1]
```diff
@@ -29,12 +29,12 @@ export default {
...
 
       if (change.deleted) {
-        return synchronizer.syncData(url, 'delete', change, { _id: change.mongoId });
+        return synchronizer.syncData(url, name, 'delete', change, { _id: change.mongoId });
       }
 
       const data = await config.shouldSync(change);
       if (data) {
-        return synchronizer.syncData(url, 'post', change, data, lastSync);
+        return synchronizer.syncData(url, name, 'post', change, data, lastSync);
```
Comment (daneryl) [inline app/api/sync/syncWorker.js:37]: @RafaPolit i think syncData already had too many arguments before, now five, i think the name can be easily removed if we move the responsibility of updating the lastSync from syncData here after each syncData call, what do you think.

**Review by daneryl — CHANGES_REQUESTED** [review]
a couple of minor comments, apart from that LGTM.

i will also add that when dealing with fixtures this size (the migration ones) found out better for readability to just have them inline on the test.

File: app/api/sync/specs/syncWorker.spec.js [hunk 2]
```diff
@@ -625,13 +635,19 @@ describe('syncWorker', () => {
...
-      const [{ lastSync }] = await syncsModel.find();
-      expect(lastSync).toBe(0);
+      const syncs = await syncsModel.find();
+      expect(syncs.length).toBe(2);
+      expect(syncs).toEqual(
+        expect.arrayContaining([
+          expect.objectContaining({ name: 'slave1', lastSync: 8999 }),
+          expect.objectContaining({ name: 'slave3', lastSync: 0 }),
+        ])
+      );
```
[the remark this answers is not in the document]
Comment (RafaPolit) [reply app/api/sync/specs/syncWorker.spec.js:650]: Indeed, it was the unknown resulting order, but yes, good catch, I can sort the query.

File: app/api/sync/syncWorker.js (hunk 1 above)
Comment (RafaPolit) [reply app/api/sync/syncWorker.js:37]: My first hunch was to use the URL as the unique identifying property, but thought name was more semantic.
Still, yes, I can look into who is responsible for what and who needs to know which values.

Comment (RafaPolit) [reply app/api/sync/syncWorker.js]: I have moved them upwards as you suggested.  Still, the function had 5 args already, so I combined 3 those that 'make sense' to be a data object to an object, and left the other two as options.

I'm not really kin of "masking" the "too many args" into objects, but, in this case, it kind of made sense.  What do you think?

Comment (daneryl) [reply app/api/sync/syncWorker.js]: looks good 👍, not sure about the object either but definitely better.

**Review by daneryl — APPROVED** [review]

**Pull request merged**